src/components/tikzpictureview.ts

```diff
@@ -0,0 +1,361 @@
+import * as vscode from 'vscode'
+import * as path from 'path'
+import * as fs from 'fs'
+import * as fse from 'fs-extra'
+import * as child_process from 'child_process'
+import { tmpdir } from 'os'
+import { promisify } from 'util'
+
+import { Extension } from '../main'
+
+const removeDir = promisify(fse.remove)
+
+interface IFileTikzCollection {
+    location: string
+    tempDir: string
+    tikzPictures: IFileTikzPicture[]
+    preamble: string
+    lastChange: number
+}
+
+interface IFileTikzPicture {
+    range: vscode.Range
+    content: string
+    tempFile: string
+    lastChange: number
+}
+
+export class TikzPictureView {
+    extension: Extension
+    tikzCollections: { [filePath: string]: IFileTikzCollection } = {}
+    initalised = false
+
+    private TEMPFOLDER_NAME = 'vscode-latexworkshop'
+
+    constructor(extension: Extension) {
+        this.extension = extension
+    }
+
+    public async view(document: vscode.TextDocument, range: vscode.Range) {
+        if (!this.initalised) {
+            await this.cleanupTempDir()
+            if (!fs.existsSync(path.join(tmpdir(), this.TEMPFOLDER_NAME))) {
+                await fs.mkdirSync(path.join(tmpdir(), this.TEMPFOLDER_NAME))
+            }
+            this.initalised = true
+        }
+
+        const fileTikzCollection = await this.getFileTikzCollection(document)
+
+        const tikzPicture = await this.getTikzEntry(fileTikzCollection, {
+            range,
+            content: document.getText(range)
+        })
+
+        const preambleStatus = await this.checkPreamble(fileTikzCollection)
+
+        if (preambleStatus === 'stable') {
+            this.updateTikzPicture(tikzPicture)
+        }
+    }
+
+    public async onFileChange(
```

Comment:
Too long. Split this function.

Reply:
See 86daf0ec53d032b66a5fa7a69bfbd9845ab2cdc2

Reply:
Aside from the big for loop, there are only ~20 lines, and I don't really see the point in extracting the for loop into its own function. Would you have any recommendations on how to split up the function in a meaningful way?

Did notice an issue with where I inserted some lines, which is fixed in e13c743176d54a6316db6dc93eb48a768896356d

Reply:
I would extract the code block following the line, `// tikzpicture modified`.

Reply:
That seems sensible. I've moved that out into an inner-function. Don't think it makes sense for it to exist in the main module. Readability is definately better if you fold that function away. See ddf871983f0d74b038caffd72a56d23d1b710e3f